lib/slapi/helpers/base.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Client Helpers for Slapi Class
+# Its main functions are to:
+#  1. Route Messages to Client
+#     - Standard Text Chat
+#     - Emote Chat
+#     - Attachment Chat
+#  2. Set response channel based on where request type
+#     - Help request response in DM if set in Bot Config
+#     - No channel set for API requests (debug level only message)
+#     - Set to same channel as request
+#  3. Set Chat listener pre-fix base on channel
+#     - Listen for @bot, bot, or bot id as initiator if normal channel
+#     - Above plus any line that starts with a listener if direct message
+class Slapi
+  def self.chat(data = nil, channel = nil, attachment)
+    @client.web_client.chat_postMessage(
+      channel: channel ? channel : channel_set(data),
+      as_user: true,
+      attachments: [attachment]
+    )
+  end
+
+  def self.chat_me(data = nil, channel = nil, attachment)
+    @client.web_client.chat_meMessage(
+      channel: channel ? channel : channel_set(data),
+      attachments: [attachment]
+    )
+  end
+
+  def self.channel_set(data)
+    # Set channel to post based on dm_user option
+    if data.empty?
+      @logger.debug("Channel request wasn't given any data")
+      nil
```

Comment:
it has to be this way, I had to add this in. I can't be sure why at the moment, I'll have to go back through commit and runt he code without it again.